icu4c/source/i18n/dangical.cpp

```diff
@@ -103,32 +103,41 @@ const char *DangiCalendar::getType() const {
  * 1898-1911: GMT+8 
  * 1912-    : GMT+9 
  */
-static void U_CALLCONV initDangiCalZoneAstroCalc(void) {
-    U_ASSERT(gDangiCalendarZoneAstroCalc == NULL);
+static void U_CALLCONV initDangiCalZoneAstroCalc(UErrorCode &status) {
+    U_ASSERT(gDangiCalendarZoneAstroCalc == nullptr);
     const UDate millis1897[] = { (UDate)((1897 - 1970) * 365 * kOneDay) }; // some days of error is not a problem here
     const UDate millis1898[] = { (UDate)((1898 - 1970) * 365 * kOneDay) }; // some days of error is not a problem here
     const UDate millis1912[] = { (UDate)((1912 - 1970) * 365 * kOneDay) }; // this doesn't create an issue for 1911/12/20
-    InitialTimeZoneRule* initialTimeZone = new InitialTimeZoneRule(UNICODE_STRING_SIMPLE("GMT+8"), 8*kOneHour, 0);
-    TimeZoneRule* rule1897 = new TimeArrayTimeZoneRule(UNICODE_STRING_SIMPLE("Korean 1897"), 7*kOneHour, 0, millis1897, 1, DateTimeRule::STANDARD_TIME);
-    TimeZoneRule* rule1898to1911 = new TimeArrayTimeZoneRule(UNICODE_STRING_SIMPLE("Korean 1898-1911"), 8*kOneHour, 0, millis1898, 1, DateTimeRule::STANDARD_TIME);
-    TimeZoneRule* ruleFrom1912 = new TimeArrayTimeZoneRule(UNICODE_STRING_SIMPLE("Korean 1912-"), 9*kOneHour, 0, millis1912, 1, DateTimeRule::STANDARD_TIME);
-    UErrorCode status = U_ZERO_ERROR;
-    RuleBasedTimeZone* dangiCalZoneAstroCalc = new RuleBasedTimeZone(UNICODE_STRING_SIMPLE("KOREA_ZONE"), initialTimeZone); // adopts initialTimeZone
-    dangiCalZoneAstroCalc->addTransitionRule(rule1897, status); // adopts rule1897
-    dangiCalZoneAstroCalc->addTransitionRule(rule1898to1911, status);
-    dangiCalZoneAstroCalc->addTransitionRule(ruleFrom1912, status);
+    LocalPointer<InitialTimeZoneRule> initialTimeZone(new InitialTimeZoneRule(
+        UnicodeString(u"GMT+8"), 8*kOneHour, 0), status);
```

Comment:
optional: you should be able to simplify `UnicodeString(u"...")` to just `u"..."` because of the implicit constructor-from-`char16_t*`

Reply:
That constructor is marked explicit in library code, so it doesn't work.